lib/teleterm/apiserver/handler/handler_device_trust.go

```diff
@@ -0,0 +1,39 @@
+// Teleport
+// Copyright (C) 2024 Gravitational, Inc.
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU Affero General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU Affero General Public License for more details.
+//
+// You should have received a copy of the GNU Affero General Public License
+// along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+package handler
+
+import (
+	"context"
+
+	api "github.com/gravitational/teleport/gen/proto/go/teleport/lib/teleterm/v1"
+	"github.com/gravitational/teleport/lib/teleterm/api/uri"
+	"github.com/gravitational/trace"
+)
+
+func (s *Handler) AuthenticateWebDevice(ctx context.Context, req *api.AuthenticateWebDeviceRequest) (*api.AuthenticateWebDeviceResponse, error) {
+	clusterURI, err := uri.Parse(req.GetRootClusterUri())
+	if err != nil {
+		return nil, trace.Wrap(err)
+	}
+
+	daemonResponse, err := s.DaemonService.AuthenticateWebDevice(ctx, clusterURI, req)
+	if err != nil {
+		return nil, trace.Wrap(err)
+	}
+
+	return &api.AuthenticateWebDeviceResponse{ConfirmationToken: daemonResponse.ConfirmationToken}, nil
```

Comment:
Why the re-wrap? Could we do this:

```suggestion
	return resp, nil
```